californium-core/src/main/java/org/eclipse/californium/core/network/stack/CoapUdpStack.java

```diff
@@ -86,23 +86,35 @@
 	 */
 	public CoapUdpStack(final NetworkConfig config, final Outbox outbox) {
 		super(outbox);
+		Layer layers[] = new Layer[] {
+				createExchangeCleanupLayer(config),
+				createObserveLayer(config),
+				createBlockwiseLayer(config),
+				createReliabilityLayer(config)};
+
+		setLayers(layers);
+	}
+
+	protected Layer createExchangeCleanupLayer(NetworkConfig config) {
```

Comment:
FMPOV, my solution with `getLayer()` instead of this set of `creates` is more flexible.

Reply:
Why I prefer protected `creates` instead of protected `getLayer` :

1) This allow to easily change  implementation of one layer.
2) This allow to easily change order of layer (in the constructor).
3) This allow to create public getter without the "not so elegant" for+instance loop.

I don't really see the case where this is more flexible. Maybe when we add a new Layer ?
But I suppose if you add a new layer in CoapUdpStack, it's obvious you should use the same `create` pattern. 

Maybe I missed something ?